src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write code here
+const listItems = [...document.querySelectorAll('li')];
+const employeesList = document.querySelector('ul');
+
+function convertToNumber(string) {
+  const salaryNumber = string.replace('$', '').replace(',', '');
+
+  return +salaryNumber;
+}
+
+function sortBySalary(items) {
+  items.sort(
+    (a, b) =>
+      convertToNumber(b.dataset.salary) - convertToNumber(a.dataset.salary),
+  );
+}
+
+function updateList(list, items) {
+  list.innerHTML = '';
+  items.forEach((item) => list.appendChild(item));
+}
+
+// function getEmployees(items) {
+//   return items.map((item) => ({
+//     name: item.textContent.trim(),
+//     position: item.dataset.position,
+//     salary: item.dataset.salary,
+//     age: item.dataset.age,
+//   }));
+// }
+
+sortBySalary(listItems);
+updateList(employeesList, listItems);
+// getEmployees(listItems);
+
+// eslint-disable-next-line no-unused-vars
+const newEmployeesList = updateList;
```

Comment:
This line of code assigns the function updateList to the variable newEmployeesList. However, newEmployeesList is not used anywhere in the rest of the code. If this is intentional, please ensure to use this new reference to the function where needed. If not, it's better to remove this line to keep the code clean and avoid confusion.